src/components/AddTodoForm/AddTodoForm.tsx

```diff
@@ -0,0 +1,116 @@
+import { FormEvent, useState } from 'react';
+import { AddTodoFormValues, Todo, User } from '../../types/types';
+import {
+  DEFAULT_VALUES,
+  MISSING_TITLE_ERROR,
+  MISSING_USER_ERROR,
+} from './constants';
+
+type Props = {
+  users: User[];
+  nextId: number;
+  onAddTodo: (todo: Todo) => void;
+};
+
+export const AddTodoForm: React.FC<Props> = ({ users, nextId, onAddTodo }) => {
+  const [formValues, setFormValues] =
+    useState<AddTodoFormValues>(DEFAULT_VALUES);
+  const [titleError, setTitleError] = useState<null | string>(null);
+  const [userError, setUserError] = useState<null | string>(null);
+
+  const handleChange = (
+    field: keyof AddTodoFormValues,
+    value: string | number,
+  ) => {
+    setFormValues(prevValues => ({ ...prevValues, [field]: value }));
+
+    switch (field) {
+      case 'title':
+        setTitleError(null);
+        break;
+      case 'userId':
+        setUserError(null);
+        break;
+      default:
+        break;
+    }
+  };
+
+  const reset = () => {
+    setFormValues(DEFAULT_VALUES);
+    setTitleError(null);
+    setUserError(null);
+  };
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+
+    const { title, userId } = formValues;
+    const isFormValid = title && userId;
+
+    if (!isFormValid) {
+      setTitleError(title.trim() ? null : MISSING_TITLE_ERROR);
+      setUserError(userId ? null : MISSING_USER_ERROR);
```

Comment:
u can use `&&` operator instead ternary. use enum here)
```suggestion
      setTitleError(!title.trim() && TypeError.MISSING_TITLE_ERROR);
      setUserError(!userId && TypeError.MISSING_USER_ERROR);
```

Reply:
I really like the idea of having enum, totally forgot about them. But I'm not quite sure I can do it with &&. We will assign an empty string or an error in the first case, and  0 or an error in the second.
Neither the empty string nor 0 matches the default value of our enam. The empty string does, theoretically, but TypeScript will not allow it.

So I'd leave it that way:
```
      setTitleError(
        title.trim() ? ErrorType.DEFAULT : ErrorType.MISSING_TITLE_ERROR,
      );
      setUserError(userId ? ErrorType.DEFAULT : ErrorType.MISSING_USER_ERROR);
```